### ISSUE TITLE
How to create a Walled Garden in a range of ip's

### ISSUE THREAD

**Issue by SergioDG-YCC** [nodogsplash/nodogsplash#376]
Now that I have been able to update NDS and OpenWrt to the latest versions, I have the problem of allowing the login, for certain dns or ip's, before authentication. The documentation is not very clear about the method, if it is configured in etc / config or user.firewall using iptables.

**Comment by bluewavenet** [comment]
NDS has the ability to "Wall Garden" a list of IPv4 / port combinations.
In the OpenWrt config you add a line for each ip/port combination eg:
`list preauthenticated_users 'allow tcp port 443 to 216.58.204.78'`

If you have a large or dynamic list, you can use ipset. This was added to NDS some years ago but unfortunately this has been forgotten and is no longer supported in the config.
It should be reinstated and I will put it on the todo list!
However if you need it, it might be possible to manually add to the init.d, although this might not work due to other code changes since it was added to NDS - can be fixed though.

**Comment by bluewavenet** [comment]
It should be noted that FAS automatically allows preauthenticated client access to fasremoteip:fasport

**Comment by SergioDG-YCC** [comment]
This is the original factory config. Really, this pre-authenticated list worked for them?
`
config nodogsplash
option redirecturl 'http://www.qq.com'
option webroot '/etc/nodogsplash/htdocs'
option gatewayinterface 'br-lan'
option gatewayname 'AD SYSTEM'
option maxclients '250'
option clientidletimeout '120'
option clientforcetimeout '120'
list authenticated_users 'allow all'
list preauthenticated_users 'allow tcp port 80 to 192.168.0.0/16'
list preauthenticated_users 'allow tcp port 80 to jd.com'
list preauthenticated_users 'allow tcp port 80 to kkk.com'
list users_to_router 'allow tcp port 22'
list users_to_router 'allow tcp port 23'
list users_to_router 'allow tcp port 53'
list users_to_router 'allow udp port 53'
list users_to_router 'allow udp port 67'
list users_to_router 'allow tcp port 80'
list fw_mark_authenticated '520'
list fw_mark_trusted '510'
list fw_mark_blocked '500'
option enabled '0'
list allow_site 'jd.com'
list allow_site 'kkk.com'
`

**Comment by bluewavenet** [comment]
You mean this was on the router in the manufacturer's firmware?  Firewall rules cannot specify domains and allow_list does not exist. So no, this will not work as indeed you found out when you tried it.

**Comment by SergioDG-YCC** [comment]
Now you understand my dilemma? =D
I try to create a range of ip's with "list preauthenticated_users' allow tcp port 443 to ...". Something like: "... 172.217.0.0/24" But I think it has not worked

**Comment by bluewavenet** [comment]
The ipset config option is probably what you need, it can be automated as an option in dnsmasq (but I have not tried it).

**Comment by SergioDG-YCC** [comment]
Thanks Bluewavenet !. For the moment I managed to fix it with the files /etc/host and /etc/config/dhcp. Indicating an IP specifies a specific url. Ex:
216.58.222.40 www.googletagmanager.com
172.217.28.206 www.google-analytics.com
And it has worked. This is momentary, until our business partner migrates their advertising to a single server. We have already noticed accesses by the same ip to:
   http://clients1.google.com/generate_204
   http://captive.apple.com/hotspot-detect.html
What creates a false connection to the internet, the customers do not see the advertising or the hotspot, and they do not have a real connection to the internet. So it is necessary to dispense with these ip's of google.
